### PR TITLE
Update and switch to Mongoose MongoDB driver

### LINKED ISSUE #369: update mongoDB Driver
our current 3.5.9 is about a year old

### PR DESCRIPTION
**Pull request by suneettipirneni**
This switches from using the `mongodb` driver to using the more popular and versatile `mongoose` driver. If you want to read about it's benefits you can refer to the [docs ](https://mongoosejs.com/).

Some notable usage changes:
- You don't have to fetch collections from the `storageService` anymore. Instead mongoose uses models that can be accessed everywhere in the project.
- Mongoose uses schemas to define how data is structured within the database.
- You don't need to call `.toArray()` on results, mongoose takes care of that for you.
- All fetched values inherit from `Document` which means you don't need to manual assign `_id` or any other boilerplate properties to the returned values.

Closes #369

### CODE REVIEW

File: src/schemas/games.schema.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+});
+
+const gameLeaderboardSchema = new Schema({
+    userId: String,
+    guildId: String,
+    games: [gameUserSchema],
+});
+
+export const TTTLeaderboardModel = mongoose.model<GameLeaderBoardDocument>('tttLeaderboard', gameLeaderboardSchema, 'tttLeaderboard');
+export const C4LeaderboardModel = mongoose.model<GameLeaderBoardDocument>('connectFourLeaderboard', gameLeaderboardSchema, 'connectFourLeaderboard');
```
Comment (c650) [inline src/schemas/games.schema.ts:17]: newline

File: src/schemas/moderation.schema.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// Moderation Warnings.
+
+const moderationWarningSchema = new Schema({
+    user: String,
+    guild: String,
+    date: Date,
+    reportId: { type: Schema.Types.ObjectId, required: false },
+}, { collection: 'modwarnings' });
+
+export const ModerationWarningModel = mongoose.model<Moderation.ModerationWarningDocument>('modwarnings', moderationWarningSchema);
```
Comment (c650) [inline src/schemas/moderation.schema.ts:39]: newline

fix ur vs settings pls

**Comment by c650** [comment]
thanks for doing this. ill leave approval to @tanndlin

File: src/schemas/moderation.schema.ts (hunk 2 above)
Comment (suneettipirneni) [reply src/schemas/moderation.schema.ts:39]: Looks like this needs an eslint rule

**Comment by suneettipirneni** [comment]
As with my other pr's git history is borked bc of the squash thing last night.

File: src/schemas/ta.schema.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import { TADocument } from '../app/plugins/ta.plugin';
+const { Schema } = mongoose;
+
+const classTASchema = new Schema({
+    userID: String,
+    chanID: String,
+    guildID: String,
+});
+
+export const ClassTAModel = mongoose.model<TADocument>('classTAs', classTASchema);
```
Comment (tanndlin) [inline src/schemas/ta.schema.ts:11]: Combine this and the pins, for a `class.schema` or something

**Review by tanndlin — APPROVED** [review]

**Comment by suneettipirneni** [comment]
Can I get another review 👉👈🥺

**Review by tanndlin — APPROVED** [review]

**Review by Armintron — APPROVED** [review]
Looks good to me
Especially admire not having to manually add collections to `storage,service.ts` 👍

**Pull request merged**